compiler-core/src/javascript/pattern.rs

```diff
@@ -343,12 +343,31 @@ impl<'module_ctx, 'expression_gen, 'a> Generator<'module_ctx, 'expression_gen, '
                 Ok(())
             }
 
-            Pattern::VarUsage { location, .. } | Pattern::BitString { location, .. } => {
-                Err(Error::Unsupported {
-                    feature: "Bit string matching".to_string(),
-                    location: *location,
-                })
+            Pattern::BitString { segments, .. } => {
+                use BitStringSegmentOption as Opt;
+
+                self.push_bitstring_length_check(subject.clone(), segments.len(), false);
+                self.push_string("buffer");
+                for (index, segment) in segments.iter().enumerate() {
+                    let _ = match segment.options.as_slice() {
+                        [] | [Opt::Int { .. }] => Ok(()),
+                        _ => Err(Error::Unsupported {
+                            feature: "This bit string segment option in patterns".to_string(),
+                            location: segment.location,
+                        }),
+                    }?;
+                    self.push_int(index);
+                    self.traverse_pattern(subject, &segment.value)?;
+                    self.pop();
+                }
+                self.pop();
+
+                Ok(())
             }
+            Pattern::VarUsage { location, .. } => Err(Error::Unsupported {
+                feature: "Bit string matching".to_string(),
```

Comment:
This is the same error as this pattern feature had before. maybe it's not so clear now but I'm not even sure how this would happen so potentially will leave improving error message till a bit later in the process.